### ISSUE TITLE
Customize the formatting layer

### ISSUE THREAD

**Issue by rustyconover** [ymgyt/tracing-cloudwatch#38]
I want to thank you for creating such a useful crate—it's been incredibly helpful for my project!

I was wondering if you'd be open to adding a feature that allows the formatting layer to be passed as a parameter? This would make it possible for me to log in JSON format to CloudWatch Logs instead of using free-form text.

Thanks again for your hard work on this crate, and I’d love to hear your thoughts!

Best,
Rusty

**Comment by ymgyt** [comment]
Thank you for your suggestion and kind words.

In fact, I had been considering providing an API that allows passing a [`fmt_layer`](https://github.com/ymgyt/tracing-cloudwatch/blob/14228194fdd355d1156e4ffa2200740ba90a6653/src/layer.rs#L47). Currently, we provide several options to customize the `fmt_layer`, but I believe that allowing users familiar with `tracing_subscriber` to pass their own `fmt_layer` would make for a more user-friendly API.

I’m a bit busy this week, but I might be able to address this and release an update next week.

**Comment by rustyconover** [comment]
Hello @ymgyt,

I greatly appreciate your help allowing the `fmt_layer` to be supplied. While I’m comfortable working with Rust, I'm not as familiar with the `tracing` crate hierarchy and the associated trait constraints needed to implement the modifications in your crate.

Please take your time.

Rusty

**Comment by ymgyt** [comment]
@rustyconover 

I have added the `with_fmt_layer()` method and released v0.2.0.
By doing the following, you should be able to output logs in JSON format.

```rust
tracing_subscriber::registry::Registry::default()
    .with(tracing_cloudwatch::layer()
        .with_client(/* ... */)
        .with_fmt_layer(
            tracing_subscriber::fmt::layer()
                .json()
                .without_time()
        )
    )
    .init();
```
I will close this issue, but feel free to reopen it if you encounter any problems.

**Comment by rustyconover** [comment]
Thank you for adding this option, I will test with it.
